lib/Renard/Curie/App.pm

```diff
@@ -46,6 +46,43 @@ sub setup_window {
 	$self->builder->connect_signals;
 }
 
+sub setup_keybindings {
+	my ($self) = @_;
+
+	$self->builder->get_object('main-window')->
+		signal_connect( key_press_event => \&key_pressed, $self );
+}
+
+sub key_pressed {
+	my ($window, $event, $self) = @_;
+
+	if($event->keyval == Gtk3::Gdk::KEY_Page_Down){
+		$self->page_document_component->set_current_page_forward;
+	} elsif($event->keyval == Gtk3::Gdk::KEY_Page_Up){
+		$self->page_document_component->set_current_page_back;
+	} elsif($event->keyval == Gtk3::Gdk::KEY_Up){
+		my $current = $self->page_document_component->scrolled_window->get_vadjustment();
+		my $adjustment = $current->get_value() - $current->get_step_increment();
+		$current->set_value($adjustment);
+		$self->page_document_component->scrolled_window->set_vadjustment($current);
```

Comment:
```diff
+		$self->page_document_component->scrolled_window->set_vadjustment($current); 
```

This line (and the others below) are not necessary. The GtkAdjustment object is a reference, not a value.